src/view.c

```diff
@@ -648,6 +648,11 @@ view_minimize(struct view *view, bool minimized)
 	struct view *root = view_get_root(view);
 	_minimize(root, minimized);
 	minimize_sub_views(root, minimized);
+
+	/* Enable top-layer when full-screen views are minimized */
+	if (view->output) {
+		desktop_update_top_layer_visiblity(view->server);
+	}
```

Comment:
Nice catch, thanks for the PR!

Can you add a check for fullscreen here (e.g. `view->fullscreen && view->output`)?
Otherwise we recalculate the visibility on every unrelated minimize event.

Reply:
Done.